lib/matplotlib/tests/test_mathtext.py

```diff
@@ -364,3 +364,13 @@ def test_mathtext_to_png(tmpdir):
     mt = mathtext.MathTextParser('bitmap')
     mt.to_png(str(tmpdir.join('example.png')), '$x^2$')
     mt.to_png(io.BytesIO(), '$x^2$')
+
+
+@image_comparison(baseline_images=['mathtext_fontset_image.png'],
+                  savefig_kwarg={'dpi': 40})
+def test_math_fontfamily():
+    fig = plt.figure(figsize=(10, 3))
+    fig.text(0.2, 0.7, r"$\mathit{This\ text\ should\ have\ one\ font} $",
+             size=24, math_fontfamily='dejavusans')
+    fig.text(0.2, 0.3, r"$\mathit{This\ text\ should\ have\ another} $",
```

Comment:
I was about to approve, but what the heck is going on with the baseline in the actual image for this font?

Reply:
Math text is italic by default, right? While implementing my changes I tested different font styles (including the redundant `\mathit{}`) to be sure they were working properly and I must have forgotten to remove this part. My apologies :)

Reply:
I wasn't worried about that, but rather the actual text comes out all crooked wrt the baseline.  I guess I feel that tests should *look* right so we can be sure they are testing correct behaviour.  
<img width="443" alt="fonts" src="https://user-images.githubusercontent.com/1562854/91072908-175aea00-e5ef-11ea-9801-b8d2a2cd664d.png">

Reply:
I will look onto that, thx for pointing it out.

Reply:
I haven't looked carefully but this *may* just be a very old bug, hinted at in https://github.com/matplotlib/matplotlib/pull/5414#issue-49863863 (the "wiggly" baselines).  IIRC I have a fix somewhere for that, but as usual it never made it even to the PR stage because of course that involves updating all baselines...

Reply:
Looking at other baseline images it seems this is a very common behavior, in accordance to what anntzer stated. Here is one example:
![image](https://user-images.githubusercontent.com/3846018/91084399-53f40900-e622-11ea-8de3-90bf7e2c2441.png)

I will try to reproduce the test in earlier versions of `matplotlib` to make sure this is not due to my changes. Additionally I can change the test to make the wiggling less noticeable but it may not be possible to get rid of it completely.

Reply:
I don't think its your changes.  OTOH, does all our math look like this or is it just the fact you are calling a sentence?

Reply:
Yes, it can occur with normal math, but it is only noticeble when there are a lot of letters together, here is one example from `baseline_images`:
![image](https://user-images.githubusercontent.com/3846018/91109351-15c20e00-e651-11ea-9194-cb1f5eec6aad.png)
The probable reason there is not a lot of complains about this is because it's hard to reproduce, if you put on an axis the wiggling will vanish:
![image](https://user-images.githubusercontent.com/3846018/91108390-59674880-e64e-11ea-8fe3-9b5ce6d081ff.png)
If you try to reproduce the same code outside the test environment, no wiggling will occur either:
![image](https://user-images.githubusercontent.com/3846018/91108551-a3502e80-e64e-11ea-912f-d0bc61bf573d.png)

It might be some setup done by the testing environment, but that's all the info I gathered so far.

Reply:
So, can we just add the axes?  I don't mean to drag you into a really old bug...

Reply:
I choose a new font that wont display the 'wiggly' behavior. This way the axes wont be needed and the image size will be kept small.

Reply:
Our tests use weird font settings in order to avoid regenerating images. Hopefully that can be dropped soon.